Formula/l/localai.rb

```diff
@@ -0,0 +1,59 @@
+class Localai < Formula
+  include Language::Python::Virtualenv
+
+  desc ":robot: The free, Open Source OpenAI alternative"
+  homepage "https://localai.io"
+  url "https://github.com/mudler/LocalAI/archive/refs/tags/v2.15.0.tar.gz"
+  sha256 "690ee0facf5450fbf9ff886ac20c6e36c68face626e2d58014cc7ccbaa330ac0"
+  license "MIT"
+
+  depends_on "abseil" => :build
+  depends_on "cmake" => :build
+  depends_on "go" => :build
+  depends_on "grpc" => :build
+  depends_on "protobuf" => :build
+  depends_on "protoc-gen-go" => :build
+  depends_on "protoc-gen-go-grpc" => :build
+  depends_on "python-setuptools" => :build
+  depends_on xcode: :build
+
+  depends_on "python@3.12"
+  depends_on "wget"
+
+  resource "grpcio-tools" do
+    url "https://files.pythonhosted.org/packages/9f/30/cd31c3a04814eb880d5e78cea768240c92fb5adaa158814c2b166356a0c6/grpcio_tools-1.64.0.tar.gz"
+    sha256 "fa4c47897a0ddb78204456d002923294724e1b7fc87f0745528727383c2260ad"
+  end
+
+  def install
+    xcode_select_path = `xcode-select --print-path`
+
+    if xcode_select_path.include?("CommandLineTools")
+      puts "Your xcode-select PATH is pointing to #{xcode_select_path}"
+      puts "In order to run the build you will need to point to xcode, by running the following command:"
+      puts "sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer"
+      exit 1
+    end
+
```

Comment:
```suggestion
```
You don't need this. This is handled by `brew`.

Reply:
@carlocab I originally didn't had this in the formula but it wasn't enough. At least from my experimentes, brew only checked that xcode was there, but it didn't really validate for the xcode-select path, so it was happy to run only witht the CommandLineTools while this isn't enough for LocalAI buildling

Reply:
Did you have
```ruby
depends_on xcode: :build
```
when your build failed?

Reply:
@carlocab yes indeed, and it effectively told me to install xcode via the apple store, which I did. then I was trying to build but it was failing missing metal which only is in the path once it is set to `/Applications/Xcode.app/Contents/Developer`

Reply:
That's unexpected, especially since our macOS runners should have `/Library/Developer/CommandLineTools` as their default `xcode-select` path.

Please remove this anyway -- if indeed it isn't handled in `brew`, then the fix should be implemented in `brew` and not in particular formulae.